### PR TITLE
[Feature] support any/all_match() for arrays

### PR DESCRIPTION
**Pull request by fzhedu**
Fixes #issue
support any/all_match() for arrays.
maps can use it by rewriting `any/all_match((k,v)->k>v,map)` to `any/all_match(map_values(transform_values((k,v)->k>v, map)))`.

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [x] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 46 / 47 (97.87%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|46|47|97.87%|[162]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 46 / 47 (97.87%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|46|47|97.87%|[162]|

**Comment by stdpain** [comment]
dup with ArrayHasImpl?

**Comment by fzhedu** [comment]
> dup with ArrayHasImpl?

no, `ArrayHasImpl` works for two array inputs, not an array with an element.

**Review by liuyehcf — APPROVED** [review]

**Comment by liuyehcf** [comment]
LGTM

**Review by Seaven — APPROVED** [review]

File: be/src/exprs/array_functions.tpp [hunk 1]
```diff
@@ -862,6 +862,68 @@ private:
...
+    static ColumnPtr process([[maybe_unused]] FunctionContext* ctx, const Columns& columns) {
+        return _array_match(columns);
+    }
+
+private:
+    static ColumnPtr _array_match(const Columns& columns) {
+        DCHECK(columns.size() == 1);
+        RETURN_IF_COLUMNS_ONLY_NULL(columns);
+        size_t chunk_size = columns[0]->size();
+        ColumnPtr bool_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[0]);
```
Comment (LiShuMing) [inline be/src/exprs/array_functions.tpp:881]: We always unpack the const input column? have not do some optimizations for const column?

Comment (fzhedu) [reply be/src/exprs/array_functions.tpp:881]: yes, we can optimize it later if needed.

**Review by packy92 — APPROVED** [review]

**Comment by fzhedu** [comment]
@mergify backport branch-3.0

**Pull request merged**